### PR TITLE
check form submit and to empty fileList

### PR DESCRIPTION
**Pull request by AmramAnanyan**

### CODE REVIEW

File: src/Body/Fields/File/File.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import React, { Fragment, useEffect, useRef, useState } from 'react'
+import React, { Fragment, useContext, useEffect, useRef, useState } from 'react'
 import { uploadFiles, correctFiles, ABORT_REQUEST_CONTROLLERS } from './../../../Helpers/files'
 import List from './List'
-import { Input, useField } from 'boomform'
+import { Context, Input, useField } from 'boomform'
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:4]: No need for context

File: src/Body/Fields/File/File.js [hunk 2]
```diff
@@ -15,16 +15,22 @@ const File = ({
...
     window._handleChange({ id, value: fileList.length === 0 ? null : fileList })
-  }, [fileList, cancelRequestId])
+  }, [fileList])
+
+
+  useEffect(() => {
+
+    setFileList([])
+    
+  }, [isSubmited])
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:33]: Use prettier.

**Comment by Tiko-Nazaryan** [comment]
Hey. 

First of all. You need to remove is Submitted state.
After you need to keep files in fields component using field_id. 
Set files data to boomform in files component where are located files state.

**Review by Tiko-Nazaryan — CHANGES_REQUESTED** [review]

File: src/Footer/SubmitButton/SubmitButton.js [hunk 3]
```diff
@@ -47,10 +48,13 @@ const SubmitButton = ({
   }
 
   const handleClick = (e) => {
-    e.preventDefault()
     if (canSubmit()) {
       if (formRef.current.checkValidity()) {
-        if (onSubmit) onSubmit({ state, actions })
+        e.preventDefault()
```
Comment (Tiko-Nazaryan) [inline src/Footer/SubmitButton/SubmitButton.js:53]: Return this guy to his place

File: src/Body/Fields.js [hunk 4]
```diff
@@ -56,14 +58,23 @@ const Fields = ({
...
 
+  useEffect(() => {
+    for (let i = 0; i < fields.length; i++) {
+      const field = fields[i]
+      if (field.type === 'file') {
+        fileList[field.id] = null
+      }
+    }
+  }, [fields])
+
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields.js:69]: try to remove this if everything is same keep it removed

File: src/Body/Fields.js [hunk 5]
```diff
@@ -56,14 +58,23 @@ const Fields = ({
     }
   }, [])
 
+  useEffect(() => {
+    for (let i = 0; i < fields.length; i++) {
+      const field = fields[i]
+      if (field.type === 'file') {
+        fileList[field.id] = null
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields.js:65]: Directly changed state

File: src/Body/Fields/File/Preview.js [hunk 6]
```diff
@@ -3,15 +3,15 @@ import React, { Fragment, useMemo } from 'react'
...
-  }
+  // const makeSize = (bytes) => {
+  //   // if      (bytes >= 1073741824) { bytes = (bytes / 1073741824).toFixed(2) + " GB"; }
+  //   // else if (bytes >= 1048576)    { bytes = (bytes / 1048576).toFixed(2) + " MB"; }
+  //   // else if (bytes >= 1024)       { bytes = (bytes / 1024).toFixed(2) + " KB"; }
+  //   // else if (bytes > 1)           { bytes = bytes + " bytes"; }
+  //   // else if (bytes == 1)          { bytes = bytes + " byte"; }
+  //   // else                          { bytes = "0 bytes"; }
+  //   // return bytes;
+  // }
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/Preview.js:14]: Move this to handlers

File: src/Body/Fields/File/File.js [hunk 7]
```diff
@@ -15,51 +15,81 @@ const File = ({
...
   }
 
   const acceptFiles = (files) => {
 
     const newFiles = correctFiles(files)
-    setFileList((prev) => [...prev, ...newFiles])
-
+    setFileList((prev) => {
+      for (let key in prev) {
+        if (key == id) {
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:82]: Should be ===

File: src/Body/Fields/File/File.js [hunk 8]
```diff
@@ -15,51 +15,81 @@ const File = ({
...
+  
   }
 
   const acceptFiles = (files) => {
 
     const newFiles = correctFiles(files)
-    setFileList((prev) => [...prev, ...newFiles])
-
+    setFileList((prev) => {
+      for (let key in prev) {
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:81]: Useles loop. you can just take prev[id]. And write a condition if it's exists or not if not just create one

File: src/Body/Fields/File/File.js [hunk 9]
```diff
@@ -15,51 +15,81 @@ const File = ({
...
+        if (key == id) {
+          if (prev[key]) {
+            prev[key]=[...prev[key],...newFiles]
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:84]: Need to clone state before working with him

File: src/Body/Fields/File/File.js [hunk 10]
```diff
@@ -15,51 +15,81 @@ const File = ({
...
+        return {
+          ...prev,
+          [id]: [...filePrecentage],
+        }
       })
-    })
+
+    }
+      
+
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:74]: Please use prettier

File: src/Body/Fields/File/File.js [hunk 11]
```diff
@@ -15,51 +15,81 @@ const File = ({
...
+        let filePrecentage = prev[id].map((file) => {
+          if (file.id === fileId) {
+            file.percentage = percentage
+          }
+          return file
+        })
+        return {
+          ...prev,
+          [id]: [...filePrecentage],
+        }
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:69]: If we are using only percentage === 0 which is start uploading and percentage === 100 which is file successfully uploaded. Why we don't use acceptFiles function which is file started to upload and  handleCallback which is file finished uploading ?

File: src/Body/Fields/File/File.js [hunk 12]
```diff
@@ -80,15 +110,24 @@ const File = ({
...
+      let filteredFiles = prev[id].filter((file) => {
+        return file.id !== fileId
+      })
+      return {
+        ...prev,
+        [id]: [...filteredFiles]
+      }
+    })
+
+    ABORT_REQUEST_CONTROLLERS.get(fileId)?.abort()
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:123]: Use ref for this.

**Review by Tiko-Nazaryan — CHANGES_REQUESTED** [review]

File: src/Body/Fields/File/File.js [hunk 13]
```diff
@@ -15,57 +19,71 @@ const File = ({
...
   const handleCallback = (fileId, status, responce, fileName) => {
-
     if (status === 200) {
-      setFileList((prev) => prev.map((file) => {
-        if (fileId === file.id) file.responce = responce
-        return file
-      }))
+      setFileList((prev) => {
+        const fileResponce = prev[id].map((file) => {
+          if (file.id === fileId) {
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/File.js:39]: Distract state before assign something to him

File: src/Body/Fields/File/List.js [hunk 14]
```diff
@@ -1,15 +1,15 @@
 import React, { memo } from 'react'
 import Preview from './Preview'
-const List = ({ value, handleRemove }) => {
+const List = ({ value, handleRemove, loadingState }) => {
   return value.map((file) => {
-    const { id, type, percentage, size, originalName } = file
-    console.log(file);
+    const { id, type, size, originalName } = file
+    console.log(file)
```
Comment (Tiko-Nazaryan) [inline src/Body/Fields/File/List.js:6]: Clarify with @sahak-sahakyan is this console needed in production

**Review by Tiko-Nazaryan — APPROVED** [review]

File: src/Body/Fields/File/File.js [hunk 15]
```diff
@@ -32,14 +32,18 @@ const File = ({
...
     if (status === 200) {
       setFileList((prev) => {
         const fileResponce = prev[id].map((file) => {
           if (file.id === fileId) {
-            file.responce = responce
+            return {
+              ...file,
+              size: file.size,
+              responce: { ...responce }
+            }
```
Comment (Tigran0199) [inline src/Body/Fields/File/File.js:44]: Hors arev👌

**Review by Tigran0199 — APPROVED** [review]

**Comment by sahak-sahakyan** [comment]
Chee vapshe needed chi

**Pull request closed without merging**